### PR TITLE
Fix/api checks

### LINKED ISSUE #84: The site performs no checks before trying to add games
We should have backend checks to prevent unreasonable games (Minimum players higher than maximum). This can also lead to the backend crashing if the action is not permitted

### LINKED ISSUE #90: I can remove a game without logging in
The API performs no checks before letting someone remove a game, you can remove from another organization without it complaining, not good! :cry: 
image

### PR DESCRIPTION
**Pull request by x183**
Added checks to gameRouter to resolve some of the more noticeable crashes and bugs occuring.
fixes #84, fixes #90 
image

### CODE REVIEW

File: backend/src/services/gameService.ts [hunk 1]
```diff
@@ -124,16 +124,20 @@ export const filterGames = async (filter: Filter) => {
 	});
 };
 
-export const removeGame = async (gameID: string) => {
+export const removeGame = async (gameID: string, userGroup: string) => {
```
Comment (Oscariremma) [inline backend/src/services/gameService.ts:127]: Could we have a better variable name than `userGroup` as it sounds like a group but is actually a game owner id?

File: backend/src/routers/gameRouter.ts [hunk 2]
```diff
@@ -164,6 +173,13 @@ gameRouter.post(
 		);
 
 		res.status(200).json({ message: 'Game added' });
+		} catch (e) {
+			if (e instanceof Error) {
+				res.status(400).json({ message: "Something went wrong adding the game" });
+			} else {
+				res.status(500).json({ message: "Uwu oopsie woopsie, the devs made a fucky wucky! Sowwy" });
```
Comment (Oscariremma) [inline backend/src/routers/gameRouter.ts:180]: image

**Review by Oscariremma — CHANGES_REQUESTED** [review]

File: backend/src/routers/gameRouter.ts (hunk 2 above)
Comment (x183) [reply backend/src/routers/gameRouter.ts:180]: :smiling_face_with_tear:

File: backend/src/services/gameService.ts (hunk 1 above)
Comment (x183) [inline backend/src/services/gameService.ts:127]:
```suggestion
export const removeGame = async (gameID: string, gameOwnerId: string) => {
```

File: backend/src/services/gameService.ts [hunk 3]
```diff
@@ -124,16 +124,20 @@ export const filterGames = async (filter: Filter) => {
...
 			id: gameID
 		},
 		select: {
-			borrow: true
+			borrow: true,
+			GameOwner: true
 		}
 	});
 	if (!game) throw new Error('Game not found');
+	if (game.GameOwner?.id != userGroup) {
```
Comment (x183) [inline backend/src/services/gameService.ts:138]:
```suggestion
	if (game.GameOwner?.id != gameOwnerId) {
```

File: backend/src/routers/gameRouter.ts (hunk 2 above)
Comment (GAsplund) [inline backend/src/routers/gameRouter.ts:180]: This message. This is perfection. No code in this repository will ever top the quality of this single line right here. We might as well archive the entire repo at this point, we're all done here.

**Review by Oscariremma — APPROVED** [review]

File: backend/src/routers/gameRouter.ts [hunk 4]
```diff
@@ -286,16 +302,22 @@ gameRouter.post('/filter', validateRequestBody(filterGamesSchema), async (req, r
...
- */
+ * @apiErrorExample {json} 401 Unauthorized:
+ * {
+ * 	"message": "Must be logged in to remove game"
+ * }
+*/
 gameRouter.post('/remove', async (req, res) => {
 	try {
-		await removeGame(req.body.id);
+		if (!req.user) { res.status(401).json({ message: 'Not logged in' }); return; }
```
Comment (GAsplund) [inline backend/src/routers/gameRouter.ts:312]: Documentation does not match actual return value

**Review by GAsplund — CHANGES_REQUESTED** [review]

File: backend/src/routers/gameRouter.ts (hunk 4 above)
Comment (x183) [inline backend/src/routers/gameRouter.ts:312]:
```suggestion
		if (!req.user) { res.status(401).json({ message: 'Must be logged in to remove game' }); return; }
```

Comment (x183) [reply backend/src/routers/gameRouter.ts]: Good catch :+1:

**Review by GAsplund — APPROVED** [review]
Looks good now!

**Pull request merged**